lib/oelite/compat.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python
+import os
+import fcntl
+import ctypes
+import sys
+
+__all__ = ["dup_cloexec", "open_cloexec"]
+
+# uname() -> (sysname, nodename, release, version, machine)
+sysname, _, release, _, machine = os.uname()
+
+values = dict()
+if (sysname, machine) == ("Linux", "x86_64"):
+    # these are supported since at least 2.6.24, so just set them unconditionally
+    values['O_CLOEXEC'] = 02000000
```

Comment:
I agree, that's the only sane way to write octal constants. But this was just copy-pasted directly from the kernel uapi header. If we ever move to python 3, we don't need to monkey-patch in support for os.O_CLOEXEC.